src/main/java/com/spotify/sparkey/SortHelper.java

```diff
@@ -39,12 +39,7 @@
 
   static final int ENTRY_SIZE = 40;
 
-  static final Comparator<Entry> ENTRY_COMPARATOR = new Comparator<Entry>() {
-    @Override
-    public int compare(final Entry o1, final Entry o2) {
-      return Long.signum(o1.wantedSlot - o2.wantedSlot) * 2 + Long.signum(o1.address - o2.address);
-    }
-  };
+  static final Comparator<Entry> ENTRY_COMPARATOR = (o1, o2) -> Long.signum(o1.wantedSlot - o2.wantedSlot) * 2 + Long.signum(o1.address - o2.address);
```

Comment:
Comparator.comparingLong(Entry::wantedSlot).thenComparingLong(Entry::address)

Reply:
(If that is the intention? If it's not the intention, this deserves a comment)

Reply:
Will fix

Reply:
Actually, this turned out to be a big hotspot, so I think it was supposed to be an optimization- will comment

Reply:
The `.comparingLong` approach might actually be faster

Reply:
Yes, I confirmed that with jmh - will fix!